app/models/promotional_feature_item.rb

```diff
@@ -10,4 +10,13 @@ class PromotionalFeatureItem < ApplicationRecord
   accepts_nested_attributes_for :links, allow_destroy: true, reject_if: ->(attributes) { attributes["url"].blank? }
 
   mount_uploader :image, ImageUploader
+
+  after_save :republish_organisation_to_publishing_api
+  after_destroy :republish_organisation_to_publishing_api
```

Comment:
There's a risk with this approach that any bulk actions performed on `PromotionalFeatureItem`s will trigger a barrage of republish calls to the Publishing API. This could prove problematic once we add the custom sort functionality which will end up updating every `PromotionalFeatureItem` on a `PromotionalFeature` to set its sort order.

This has hurt us in the past – for example, I remember a situation where [document attachments behaved similarly](https://docs.google.com/document/d/11zmKzkbUxlJtd8Din3Y_x0u46mPhL1lX0saQiVnuzfo/edit#) and caused a GOV.UK incident. And a [similar incident occurred](https://docs.google.com/document/d/1exAZhfDg1ZmlCQ0hmE3flhFh9EpXO3qDr-EG4a6Dz1Y/edit#) on orgs with large numbers of featured documents.

I expect this is less likely to result in incidents due to only two pages using promotional features, and neither using very many. So they won't result in huge spikes of republish jobs. However, it'd be ideal if we could find somewhere 'further up' in the code so this republishing only gets triggered once per user-performed action.

Off the top of my head, I don't have any suggestions for how/where to implement that. (Sorry!) But I'm hoping we can find something.

Reply:
Okay cool. I think we can just add it into the promotional features controller 👍

Reply:
Yeah I think moving things to controllers (or services) is preferable to using active record callbacks. Ideally we'd get rid of all the active record callbacks that trigger writes to the Publishing API.

Reply:
okay cool they're in a few other places. I can look at them over the holiday break